bin/varnishd/builtin.vcl

```diff
@@ -209,6 +209,10 @@ sub vcl_backend_response {
 	return (deliver);
 }
 
+sub vcl_backend_refresh {
+	return (merge);
+}
```

Comment:
I'm having second thoughts here, and I'm wondering whether we should generalize this subroutine to whenever there is a stale object involved.

```vcl
sub vcl_backend_refresh {
        if (obj_stale.status == 200 && beresp.status == 304) {
                return (merge);
        }
        if (beresp.status == 304) {
                return (error(503)); # until we refine error statuses
        }
        return (beresp);
}
```

Of course, with a pinch of splitting.

Reply:
@Dridi can you please motivate this suggestion? What else would you do with the stale object besides delivering it as-is (for grace mode) or use it for a refresh?

Reply:
bugwash: This makes sense.